include/envoy/stream_info/stream_info.h

```diff
@@ -116,6 +123,11 @@ class StreamInfo {
    */
   virtual void setResponseFlag(ResponseFlag response_flag) PURE;
 
+  /**
+   * @param rc_details the response code details string to set for this request.
+   */
+  virtual void setResponseCodeDetails(const std::string& rc_details) PURE;
```

Comment:
How about absl::string_view?

Some of the details I'd like to see in future include things like passing the http_errno_name from the H1 codec, and those come as char*s, so it'd be nice to avoid stringifying and then copying.

Reply:
Ok, changed the signature but I left the StreamInfo field as absl::optional<std::string> so there's still going to be a copy. Do you think I should change the field to absl::string_view as well? For that to be safe we'd need to be sure that the actual storage of the string will outlive the StreamInfo object.